packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -287,11 +287,58 @@ function checkFunctionExpressionReturnType(
   checkFunctionReturnType(node, options, sourceCode, report);
 }
 
+/**
+ * Check whether any ancestor of the provided function has a valid return type.
+ */
+function ancestorHasReturnType(node: FunctionNode): boolean {
+  let ancestor = node.parent;
+
+  if (ancestor?.type === AST_NODE_TYPES.Property) {
+    ancestor = ancestor.value;
+  }
+
+  // if the ancestor is not a return, then this function was not returned at all, so we can exit early
+  const isReturnStatement = ancestor?.type === AST_NODE_TYPES.ReturnStatement;
+  const isBodylessArrow =
+    ancestor?.type === AST_NODE_TYPES.ArrowFunctionExpression &&
+    ancestor.body.type !== AST_NODE_TYPES.BlockStatement;
+  if (!isReturnStatement && !isBodylessArrow) {
+    return false;
+  }
+
+  while (ancestor) {
+    switch (ancestor.type) {
+      case AST_NODE_TYPES.ArrowFunctionExpression:
+      case AST_NODE_TYPES.FunctionExpression:
+      case AST_NODE_TYPES.FunctionDeclaration:
+        if (ancestor.returnType) {
+          return true;
+        }
+        // assume
```

Comment:
assume... what?

Reply:
this was copy pasted from the old code.
idk what it was supposed to be (I probably wrote it lol)

Reply:
I think it was just a mistake to remove it. Comments on this function I found are already giving some good "whys" and leaving it may confuse others(actually including me when I first saw it) thus I removed it.